app/scripts/lib/origin-check.js

```diff
@@ -25,11 +25,14 @@
 
 define([
   'lib/promise',
-  'lib/channels/iframe'
-], function (p, IFrameChannel) {
+  'lib/channels/iframe',
+  'raven'
+], function (p, IFrameChannel, Raven) {
   'use strict';
 
-  var RESPONSE_TIMEOUT_MS = 100;
+  // This timeout needs to be long enough for slow machines to able to respond to
+  // See https://github.com/mozilla/fxa-content-server/issues/2946 for details
+  var RESPONSE_TIMEOUT_MS = 5000;
```

Comment:
@TDA it's a "tested" number for now. it is doubled because the firstrun page is heavier. 
However in case of timeout we do nothing anyway and just log a message. Only those who are iframing without permission have to wait 5 seconds to get a respone back.